util/src/main/java/io/camunda/zeebe/util/startup/AbstractStartupStep.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.util.startup;
+
+import java.util.concurrent.CompletableFuture;
+
+public abstract class AbstractStartupStep<CONTEXT> implements StartupStep<CONTEXT> {
```

Comment:
:x: I wouldn't add this just yet as it's only used for testing, and it's unclear to me if this will be necessary. I expect startup step targeting actors will not need this abstract class as they can perform their own synchronization. Is this then for other steps? In the spike, I see it's used to also wrap actor steps. I'm not sure this added synchronization is required. Maybe I don't get the concept, so it might be worth if we take a bit of time and you explain to me why an abstract class which performs the guard around the startup/shutdown is required, even for actors. And if not for actors, where would be required? I guess it seems a bit premature for me at the moment.

Reply:
I agree with your caution here. I was very enthusiastic about this class at first. 

Then about midway, I felt that this class should not exist in development code but rather in the test code. And then it would be used to test whether the startup process complies with the contract for steps.

I still see a little bit of value in the handling of futures and in the finesse about if startup is still running and it completes that then the context returned by the startup process is used for shutdown. However, some of this is rather academic in nature. 

When I started implementing the spike I was not fully committed to actors yet. And I am still hesitant to commit to the actors we have. But yes, if we assume actors then this class doesn't have too much value on its own.

Let's discuss it when we talk about the next steps and how

Reply:
On second thought (weekend helps with that), I just removed it.